ros/src/computing/planning/motion/packages/waypoint_follower/include/mpc_follower/lowpass_filter.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2018-2019 Autoware Foundation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file lowpass_filter.h
+ * @brief vehicle model interface class
+ * @author Takamasa Horibe
+ * @date 2019.05.01
+ */
+
+#pragma once
+#include <cmath>
+#include <vector>
+#include <algorithm>
+
+/** 
+ * @class 2nd-order Butterworth Filter
+ * @brief filtering values
+ */
+class Butterworth2dFilter
+{
+private:
+  double y1_; //!< @brief filter coefficient calculated with cutoff frequency and sampling time
+  double y2_; //!< @brief filter coefficient calculated with cutoff frequency and sampling time
+  double u1_; //!< @brief filter coefficient calculated with cutoff frequency and sampling time
+  double u2_; //!< @brief filter coefficient calculated with cutoff frequency and sampling time
+  double a0_; //!< @brief filter coefficient calculated with cutoff frequency and sampling time
+  double a1_; //!< @brief filter coefficient calculated with cutoff frequency and sampling time
+  double a2_; //!< @brief filter coefficient calculated with cutoff frequency and sampling time
+  double b0_; //!< @brief filter coefficient calculated with cutoff frequency and sampling time
+  double b1_; //!< @brief filter coefficient calculated with cutoff frequency and sampling time
+  double b2_; //!< @brief filter coefficient calculated with cutoff frequency and sampling time
+
+public:
+  /**
+   * @brief constructor with initialization
+   * @param dt sampling time
+   * @param f_cutoff_hz cutoff frequency [Hz]
+   */
+  Butterworth2dFilter(double dt = 0.1, double f_cutoff_hz = 10.0);
+
+  /**
+   * @brief destructor
+   */
+  ~Butterworth2dFilter();
+
+  /**
+   * @brief constructor
+   * @param dt sampling time
```

Comment:
Please add `in`/`out`/`inout` to the parameters that need it as you have in `qp_solver_qpoases.h`. It clarifies further the documentation for other users

Reply:
modified documents